src/reduxForm.js

```diff
@@ -433,11 +433,25 @@ const createReduxForm =
 
         const connector = connect(
           (state, props) => {
-            const { form, getFormState, initialValues } = props
+            const { form, getFormState, initialValues, enableReinitialize, keepDirtyOnReinitialize } = props
             const formState = getIn(getFormState(state) || empty, form) || empty
             const stateInitial = getIn(formState, 'initial')
-            const initial = initialValues || stateInitial || empty
-            const values = getIn(formState, 'values') || initial
+
+            const shouldUpdateInitialValues = !deepEqual(initialValues, stateInitial) && enableReinitialize
```

Comment:
@ncphillips Thanks for working on this!
To get a small performance gain I would consider to reorder this condition:
`const shouldUpdateInitialValues = enableReinitialize && !deepEqual(initialValues, stateInitial)`